metaflow/plugins/datatools/s3/s3.py

```diff
@@ -792,6 +792,31 @@ def _head():
 
         return list(starmap(S3Object, _head()))
 
+    def delete(self, key=None, return_info=True):
```

Comment:
This API seems a little bit confusing:
  - I don't think `key=None` makes sense (or maybe is too dangerous)
  - similarly, return_info also does not make sense (there is nothing to be deleted).

One parameter that may make sense is `error_on_absent` or something like that, similar to the overwrite parameter for writes in a way.

It would also be nice to use the `_one_boto_op` function like all other functions to benefit from the same retry logic as those.

Finally, and I know this is harder but we should consider a `delete_many` and, possibly, a `delete_recursive`.

Reply:
I have added `delete_many`. Could you explain what should happen in `delete_recursive`?